test/index.test.js

```diff
@@ -112,27 +131,63 @@ describe('AdWords', function() {
           google_conversion_language: 'en',
           google_conversion_format: '3',
           google_conversion_color: 'ffffff',
-          google_conversion_label: options.events.signup,
+          google_conversion_label: options.labelMap[0].value.label,
           google_conversion_value: 0,
           google_remarketing_only: false
         });
       });
 
-      it('should support array events', function() {
-        adwords.options.events = [{ key: 'login', value: 'QbThCM_zogcQofXB0gM' }];
-        analytics.track('login');
+      it('should use mapping specific IDs if provided', function() {
+        adwords.options.labelMap = [
+          {
+            key: 'signup',
+            value: {
+              eventName: 'signup',
+              label: '-kGkCJ_TsgcQofXB0gM',
+              conversionId: 431
+            }
+          },
+          {
+            key: 'login',
+            value: {
+              eventName: 'login',
+              label: 'QbThCM_zogcQofXB0gM',
+              conversionId: 23
+            }
+          },
+          {
+            key: 'play',
+            value: {
+              eventName: 'play',
+              label: 'b91fc77f',
+              conversionId: 4512
+            }
+          }
+        ];
+        analytics.track('signup', {});
         analytics.called(window.google_trackConversion, {
-          google_conversion_id: options.conversionId,
+          google_conversion_id: 431,
           google_custom_params: {},
           google_conversion_language: 'en',
           google_conversion_format: '3',
           google_conversion_color: 'ffffff',
-          google_conversion_label: adwords.options.events[0].value,
+          google_conversion_label: options.labelMap[0].value.label,
           google_conversion_value: 0,
           google_remarketing_only: false
         });
       });
 
+      it('should use whitelisted event-specific ID when provided', function() {
+        adwords.options.whitelist = [{ value: { eventName: 'danny mcbride is funny', conversionIds: [4879235] } }];
```

Comment:
like this would mean that i have to type `danny mcbride is funny` in two separate sheets when I could let you know that i want to whitelist this event by checking a box in labelMap